### ISSUE TITLE
feature request / proposal: support for RGB colors

### ISSUE THREAD

**Issue by jchaves** [chenasraf/terminal_color_parser_dart#1]
Hi there,

As mentioned [here](https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797#rgb-colors), some terminals support "RGB" colors with (usually) the format:

| ESC Code Sequence       | Description                  |
| :---------------------- | :--------------------------- |
| `ESC[38;2;{r};{g};{b}m` | Set foreground color as RGB. |
| `ESC[48;2;{r};{g};{b}m` | Set background color as RGB. |

I've found your library and was using it for a fun hobby little project, and came across some content that uses precisely this RGB format.

In this branch you can take a look at what I've done so far to add rgb support on top of your library, and if you want, you could merge it later into yours (right now, I don't think it's ready for a pull request. I'm not really a developer, there are things that I'm not yet fully understanding...)

https://github.com/jchaves/terminal_color_parser_dart/tree/feature/rgb_support

Anyway, if you'd be interested in incorporating these "colors" into your project, let me know. Maybe you could even help me understand some bits I might be missing (TODOs are there if you are curious)

Regards, and thanks for making the library in the first place!

**Comment by chenasraf** [comment]
Sounds really good, I'm gonna take a closer look at your code in the following days :)

Thanks very much for your contribution!

If I find it only requires small changes, we'll discuss what to do here

Since there seems to be a lot of code changes there, I would like to consider a possibility to integrate this feature with the current structure - but if it's not applicable, we'll get it sorted.

Either way this is going to be implemented, I'll update you with more details soon.

**Comment by jchaves** [comment]
Awesome. And yeah, it's very much a work in progress/real mess :), but the meat of it is pretty much in the  [parser.dart->processTokenStyle](https://github.com/jchaves/terminal_color_parser_dart/blob/feature/rgb_support/lib/src/parser.dart#L154) function, called recursively until all the style/color codes have been looked at.

**Comment by chenasraf** [comment]
Hey @jchaves, sorry this took so long.

I re-organized the code in a bit more readable and maintainable way and I added it. I did my best to preserve commit credits so you should be listed as participant properly :)

I'm going to make a few small updates and re-haul the color system - it was too restrictive and contained a matrix of options regarding xterm/ANSI representations of colors, so I want to make the colors a bit more usable.

After that I'll release an official update.

Thanks for your contribution!

**Comment by jchaves** [comment]
Hey there!
(Don't worry about the time, I've been on holidays and not spending much time on anything computer-y. No worries about the "credit", either!)

Good to hear. I took a quick look, and yeah, your version looks quite more understandable than my work-in-progress mess.

While taking a quick look in my phone, I remembered I had changed also the pubspec.yml to see if it would work in my project (long story short, I wanted to add some color to an app I want to run in an old Android tablet, so I'm working on/with old versions of stuff for now).

So, maybe you should check that I didn't break anything there, just in case. 

If you can keep the low version dependencies, that would probably help me, but if not, don't worry too much.

Thanks for everything!

**Comment by chenasraf** [comment]
Yes, sadly I had to revert the pubspec changes, pub store doesn't like it when I push packages with outdated dependencies. I'm afraid you'll have to either find a way to override it manually or use your fork to keep this file different.

Really sorry about that
